File: gui/ncurses.rb
#!/usr/bin/ruby
require 'ncurses'
include Ncurses
begin
fil,col=Array.new,Array.new
Ncurses.initscr
Ncurses.cbreak
Ncurses.noecho
Ncurses.start_color
Ncurses.init_pair(1, COLOR_BLUE, COLOR_RED)
Ncurses.init_pair(2, COLOR_WHITE, COLOR_BLUE)
Ncurses.stdscr.getmaxyx(fil,col)
Ncurses.stdscr.bkgd(Ncurses.COLOR_PAIR(2))
Ncurses.mvprintw(fil[0]+10/2,col[0]/2,"holaa")
Ncurses.refresh
ventana = WINDOW.new(20,30,0,0)
ventana.box(0,0)
venta = WINDOW.new(20,30,0,40)
venta.box(0,0)
venta.wrefresh
ventana.attron(Ncurses.COLOR_PAIR(1))
ventana.mvprintw(1,1,"Inserte una letra")
ventana.attroff(Ncurses.COLOR_PAIR(1))
ventana.wrefresh
caracter = ventana.getch
ventana.mvprintw(2,2,caracter.to_s)
ventana.wrefresh
ventana.attron(Ncurses.COLOR_PAIR(1))
ventana.mvprintw(3,1,"Ahora una palabra")
ventana.attroff(Ncurses.COLOR_PAIR(1))
cadena=""
ventana.getstr(cadena)
ventana.mvprintw(4,1,cadena)
ventana.wrefresh
ventana.attron(Ncurses.COLOR_PAIR(1))
ventana.mvprintw(5,1,"Presiona intro")
ventana.attroff(Ncurses.COLOR_PAIR(1))
ventana.wrefresh
#ventana.getstr(cadena)
Ncurses.keypad(Ncurses.stdscr, true)
char=stdscr.getch
if char == KEY_ENTER or char == ?\n or char == ?\r
	venta.mvprintw(1,1,"Cambio de ventana")
	venta.wrefresh
end
if char == ?\t 
	venta.mvprintw(1,1,"tabulador1!!")
	venta.wrefresh
end
ventana.delwin
venta.delwin
#Ncurses.refresh
ensure
Ncurses.endwin
end

File: gui/ncurses.py
#!/usr/bin/env python
import sys
import os
import curses

# La idea es q haya otro tipo de visores,como arbol, y se pueda coger con una barra
# las listas deberian ser otra abstraccion


#LA representacion y el contenido es independiente
class Contenedor:
	pass


class Ventana:
	def __init__(self,ancho,alto):
		self.width=0
		self.height=0
		self.xpos=0
		self.ypos=0
		self.window=curses.newwin(ancho,alto)
	def setsize(self,x,y):
		self.window.resize(x,y)
	def setposition(self,x,y):
		self.xpos=x
		self.ypos=y
		self.window=curses.newwin(self.width,self.height,x,y)
	def refresh(self):
		self.window.refresh()

class Lista(list, Contenedor, Ventana):
	def __init__(self):
		list.__init__(self)
		Ventana.__init__(self,0,0)

	def refresh(self):
		self.window.box()
		self.window.refresh


class VentanaGeneral(Ventana):
	def __init__(self,elementoinicial,ancho,alto):
		Ventana.__init__(self,ancho,alto)
		self.elemento=elementoinicial

	def refresh(self):
		Ventana.refresh(self)
		self.elemento.setsize(30,30)
		self.elemento.setposition(self.xpos+1,self.ypos+1)
		self.elemento.refresh()
		self.window.box(20,20)
		self.window.addstr(1,1,"pepe",curses.A_BLINK)



class VentanaElemento(Ventana):
	def __init__(self,elementoinicial,ancho,alto):
		Ventana.__init__(self,ancho,alto)
		self.elemento=elementoinicial

	def refresh(self):
		Ventana.refresh(self)
		self.window.box(20,20)
		self.window.addstr(1,1,str(self.elemento),curses.A_BLINK)
		self.window.addstr(2,1,"Longitud:"+str(len(self.elemento)),curses.A_BLINK)


class Planificador:
	"""Esta clase introduce los nuevos elementos según criterios"""
	def __init__(self,mainwindow):
		self.mainwindow=mainwindow
		elementoraiz=Lista()
		self.visorgeneral=VentanaGeneral(elementoraiz,mainwindow.getmaxyx()[0],(mainwindow.getmaxyx()[1]/2))
		self.visorelemento=VentanaElemento(elementoraiz,mainwindow.getmaxyx()[0],mainwindow.getmaxyx()[1]/2)
		self.visorelemento.setposition(0,(mainwindow.getmaxyx()[1]/2))


	def begin(self):
		key = self.mainwindow.getch()
		while 1:
			if key==ord('q'):
				break
			else:
				stdscr.refresh()
				self.visorgeneral.refresh()
				self.visorelemento.refresh()
				key = stdscr.getch()



if __name__ == "__main__":
	stdscr = curses.initscr()
	curses.noecho()
	curses.cbreak()
	stdscr.keypad(1)
	principal = Planificador(stdscr)
	try:
		principal.begin()
		curses.nocbreak()
		stdscr.keypad(0)
		curses.echo()
	finally:
		curses.endwin()
